### PR TITLE
Create HTTP resource with specific input and output formats

### PR DESCRIPTION
**Pull request by Vanish404**
More information here https://github.com/pub-t/plain-destination/issues/11#issuecomment-227667642

### CODE REVIEW

**Comment by irnc** [comment]
Please add more meaningful title to this pull request, _Supertest_ does not tell what you changes do to the project. Please read https://github.com/erlang/otp/wiki/Writing-good-commit-messages

File: app.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+var request = require('request');
 var http = require('http');
 var app = express();
 var bodyParser = require('body-parser');
 
 var urlencodedParser = bodyParser.urlencoded({ extended: false });
 
 app.post('/api', urlencodedParser, function (req, res) {
-    res.sendFile(__dirname + "/" + 'example.json');
+    request({url: 'https://nominatim.openstreetmap.org/search/' + req.body.name + '?format=json'}, function (err, response,data) {
```
Comment (irnc) [inline app.js]: This line is too long, extract URL string into a variable.

File: app.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+        if(err) return next(err);
```
Comment (irnc) [inline app.js]: Please use https://www.npmjs.com/package/eslint-config-airbnb to lint you code.

File: package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
   "description": "Преобразует локальное имя объекта в координату",
   "main": "app.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "node_modules/mocha/bin/mocha",
```
Comment (irnc) [inline package.json]: `npm` would add `node_modules/.bin` to `PATH` before running command, so you can write here `{ "test": "mocha" }`

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    "extends": "airbnb",
+    "plugins": [
+        "react"
+    ]
+};
```
Comment (irnc) [inline .eslintrc.js]: Please add newline at the end of file and remove `.js` from filename.

File: test/test.js [hunk 5]
```diff
@@ -1,16 +1,30 @@
...
-            .expect({lat: "53.70177645", lon: "23.8347894179425",
-                display_name: "OldCity, 17, улица Дубко, Девятовка, Ленинский район, Hrodna, Hrodna region, 230012, Belarus"});
+          .post('/searchCoordinate')
+          .send({
+            location: {
+              accuracy: 40,
+              lat: '53.663482',
+              lon: '23.834427',
+            },
+            to: 'ул. Лиможа 27/1',
```
Comment (irnc) [inline test/test.js]: `name` would be more convenient than `to`.

File: test/test.js [hunk 6]
```diff
@@ -1,16 +1,30 @@
...
+      function (done) {
         request(app)
-            .post("localhost:7788/serchСoordinate")
-            .send({SearchString:'Oldcity', BaseСoordinate: {lat: "53.93892", lon: "28.6496"}}) //текс + координата базовая
-            .expect(200)
-            .expect({lat: "53.70177645", lon: "23.8347894179425",
-                display_name: "OldCity, 17, улица Дубко, Девятовка, Ленинский район, Hrodna, Hrodna region, 230012, Belarus"});
+          .post('/searchCoordinate')
+          .send({
+            location: {
```
Comment (irnc) [inline test/test.js]: `base` would be more convenient than `location`.

File: test/test.js [hunk 7]
```diff
@@ -1,16 +1,30 @@
...
+              lat: '53.663482',
+              lon: '23.834427',
+            },
+            to: 'ул. Лиможа 27/1',
+          })
+          .expect(200)
+          .expect({
+            lat: '53.70177645',
+            lon: '23.8347894179425',
+            display_name: 'OldCity, 17, улица Дубко, Девятовка, Ленинский район, ' +
```
Comment (irnc) [inline test/test.js]: In what universe, `ул. Лиможа 27/1` should resolve to `OldCity, 17, улица Дубко`? ;)

File: test/test.js [hunk 8]
```diff
@@ -1,16 +1,30 @@
...
+describe('transforms the name of the object in a coordinate', function () {
+  it('Input place name-string,base coordinate, return coordinate and full name point',
+      function (done) {
         request(app)
-            .post("localhost:7788/serchСoordinate")
-            .send({SearchString:'Oldcity', BaseСoordinate: {lat: "53.93892", lon: "28.6496"}}) //текс + координата базовая
-            .expect(200)
-            .expect({lat: "53.70177645", lon: "23.8347894179425",
-                display_name: "OldCity, 17, улица Дубко, Девятовка, Ленинский район, Hrodna, Hrodna region, 230012, Belarus"});
+          .post('/searchCoordinate')
```
Comment (irnc) [inline test/test.js]: `POST /searchCoordinate` is a remote procedure call, we want it to implement `GET /point`, and allow it to be cacheable for example `GET /point?name=OldCity&lat=53.66&lon=23.83`

File: test/test.js [hunk 9]
```diff
@@ -1,16 +1,30 @@
-var request = require('supertest');
-var app = require('../app');
-describe("transforms the name of the object in a coordinate", function (){
-
-    it("Input place name-string,base coordinate, return coordinate and full name point", function(done){
-
+const request = require('supertest');
+const app = require('../app');
+describe('transforms the name of the object in a coordinate', function () {
+  it('Input place name-string,base coordinate, return coordinate and full name point',
```
Comment (irnc) [inline test/test.js]: It should be `describe('noun. some name here')` + `it('should do something')`, so you describe what something **should** do.

File: package.json [hunk 10]
```diff
@@ -18,9 +20,16 @@
   "homepage": "https://github.com/pub-t/name-to-point#readme",
   "dependencies": {
     "body-parser": "^1.15.1",
-    "express": "^4.13.4"
+    "express": "^4.13.4",
+    "nconf": "^0.8.4",
+    "request": "^2.72.0"
   },
   "devDependencies": {
+    "eslint": "^2.12.0",
```
Comment (irnc) [inline package.json]: Adding eslint should be done as a separate pull request, because we could merge it independently without arguing over API and testing.

File: config/config.json [hunk 11]
```diff
@@ -0,0 +1,5 @@
+{
+  "port": 7788,
+  "address": "localhost:7788"
```
Comment (irnc) [inline config/config.json]: What does `address` mean and why we need it here?

**Comment by Vanish404** [comment]
Done, can we merge?

File: README.md [hunk 12]
```diff
@@ -1,2 +1,9 @@
 # name-to-point
-Преобразует локальное имя объекта в координату
+`Преобразует локальное имя объекта в координату`
```
Comment (irnc) [inline README.md]: Description shouldn't be monospaced.

File: README.md [hunk 13]
```diff
@@ -1,2 +1,9 @@
...
+
+### Install dependencies
+`npm install`
+
+### How to run?
+* **MacOS or Linux**
```
Comment (irnc) [inline README.md]: Why not Windows? I guess following command is multi-os.

File: test/test.js [hunk 14]
```diff
@@ -1,16 +1,34 @@
...
+            ' Гродно, Гродненская область, 230012, Беларусь',
+          })
+          .end(function (err, res) {
+            if (err) {
+              return done(err);
+            }
             done();
-        
+          });
+      });
```
Comment (irnc) [inline test/test.js]: This indentation does not follow airbnb rules.

**Comment by irnc** [comment]
@Vanish404 we could merge and work on issues after merge, as a lot of work has been done already.

**Comment by irnc** [comment]
Please fix style issues and merge.

**Comment by Vanish404** [comment]
All fixed, i merge pr
